gitops/PrivateRepositories.md

```diff
@@ -0,0 +1,111 @@
+
+## Personal Access Tokens
+
+Git repository providers such as Azure DevOps or Github will often provide the ability to generate a Personal Access Token for authenticating and accessing a repository on behalf of the user the token was issued to. These tokens, depending on the issuer, will provide the same permissions as the user they authenticate on behalf of.
+
+- For Github, If a user has access to repositories `A`, `B`, and `C`, then the personal access token issued to the user maintains the same access control.
+    - You can create a Personal Access Token for a Github user account [here](https://github.com/settings/tokens), and they last until they are revoked.
+
+- For Azure DevOps there are a number of considerations when utilizing Personal Access Tokens for repository access:
+    - A Personal Access Token can be generated for two situations, with varying selectable time spans for each token. The token types are described as follows:
+
+      - A single Organization that a user is a member of
+      - All accesible Organizations a user is a member of
+
+    - Azure DevOps provides further Access Controls to Git Repositories via Roles. At minimum, a user that must be able to fetch components hosted in private repositories must also have the Git Repository Reader access control set in the Azure DevOps Security Group they belong to. This control can be configured in the Project Settings like so:
+![Where to configure the Repository Read setting in Azure DevOps](./images/azdo_git_permission.png)
+    - You can create a Personal Access Token for an Azure DevOps User within your Organization settings. Example url for token settings, replace ${organization} with your organization name: https://dev.azure.com/${organization}/_usersSettings/tokens.
+
+After you've generated a Personal Access Token for your user account, you must configure Fabrikate (0.13.0 or higher) with knowledge on how to utilize it. Fabrikate 0.13.0 introduces a new configuration file, `access.yaml`, in which Git repositories map to the _name_ of the environment variable that holds the Personal Access Token. This layer of indirection ensures that a user does not simply encode Personal Access Tokens directly into a file.
+
+Let's refer back at the `component.yaml` file from earlier:
+
+```
+name: my-component
+method: local
+source: component-files
+subcomponents:
+  - name: private-component
+    method: git
+    source: https://github.com/contoso/my-private-repository
+```
+
+At the same level, we introduce an `access.yaml` file with the following contents:
+
+```
+https://github.com/contoso/my-private-repository: ENV_VAR_CONTAINING_PAT
+```
+
+To test that this workflow works, you may:
+  1. Set the environment variable value `ENV_VAR_CONTAINING_PAT` to a Personal Access Token for the referenced repository (https://github.com/contoso/my-private-repository).
+  2. Invoke `fab install` from the same directory as the component.
+
+## SSH Keys
```

Comment:
Maybe we should mention somewhere that SSH Keys and private repos are currently only supported in version 0.15.1.

Reply:
actually, SSH keys work on any version - since they're not fab specific

Reply:
ah right - azdo client doesn't do it correctly…

Reply:
I think this is resolved via changes